src/citrine/_rest/collection.py

```diff
@@ -134,13 +134,7 @@ def _build_collection_elements(self,
 
         """
         for element in collection:
-            try:
-                yield self.build(element)
-            except(KeyError, ValueError) as e:
-                # TODO:  This is a patch to handle deprecated predictors client side.
-                # Remove when predictors are migrated
-                warnings.warn(f"Building element skipped due to error: {e}")  # pragma: no cover
-                pass  # pragma: no cover
+            yield self.build(element)
```

Comment:
This allows that test that expects a `ValueError` to property function. We were previously catching the `ValueError` based on this `TODO` message fallback. 

I am not sure if it is safe to remove this, but just posting a version with it removed and `Property.deserailize_from_dict` throwing a `ValueError` instead of a `RuntimeError`.

Reply:
As it turns out, we cannot remove this yet. It is related to #641 , and the new `/predictor` routes are not fully intregrated to the backend yet.

Because of that, this needs to stay with the try/catch. The `properties.Union` needs to throw a value error that is caught here, and `Property.deserialize_from_dict` needs to throw  a `RuntimeError` to bypass this point.

Ugly, but that's what we do for now. I'll ticket it so we remember to clean this up at some point.